From 8b7a0e262ae5d365a50449ddad6c84db438de0f1 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <heinrich.schuchardt@canonical.com>
Date: Tue, 1 Mar 2022 12:14:31 +0100
Subject: [PATCH 121/261] commands/efi/lsefisystab: Short text
 EFI_IMAGE_SECURITY_DATABASE_GUID

The EFI_IMAGE_SECURITY_DATABASE_GUID is used for the image execution
information table (cf. UEFI specification 2.9, 32.5.3.1 Using The Image
Execution Information Table).

The lsefisystab command is used to display installed EFI configuration
tables. Currently it only shows the GUID but not a short text for the
table.

Provide a short text for the EFI_IMAGE_SECURITY_DATABASE_GUID.

Signed-off-by: Heinrich Schuchardt <heinrich.schuchardt@canonical.com>
Reviewed-by: Daniel Kiper <daniel.kiper@oracle.com>
---
 grub-core/commands/efi/lsefisystab.c | 1 +
 include/grub/efi/api.h               | 5 +++++
 2 files changed, 6 insertions(+)

diff --git a/grub-core/commands/efi/lsefisystab.c b/grub-core/commands/efi/lsefisystab.c
index 456198e5f..412adbc2c 100644
--- a/grub-core/commands/efi/lsefisystab.c
+++ b/grub-core/commands/efi/lsefisystab.c
@@ -44,6 +44,7 @@ static const struct guid_mapping guid_mappings[] =
     { GRUB_EFI_DXE_SERVICES_TABLE_GUID, "DXE SERVICES"},
     { GRUB_EFI_HCDP_TABLE_GUID, "HCDP"},
     { GRUB_EFI_HOB_LIST_GUID, "HOB LIST"},
+    { GRUB_EFI_IMAGE_SECURITY_DATABASE_GUID, "IMAGE EXECUTION INFORMATION"},
     { GRUB_EFI_LZMA_CUSTOM_DECOMPRESS_GUID, "LZMA CUSTOM DECOMPRESS"},
     { GRUB_EFI_MEMORY_TYPE_INFORMATION_GUID, "MEMORY TYPE INFO"},
     { GRUB_EFI_MPS_TABLE_GUID, "MPS"},
diff --git a/include/grub/efi/api.h b/include/grub/efi/api.h
index 6c8d06e15..d73e8198e 100644
--- a/include/grub/efi/api.h
+++ b/include/grub/efi/api.h
@@ -284,6 +284,11 @@
     { 0x9a, 0x3a, 0x00, 0x90, 0x27, 0x3f, 0xc1, 0x4d } \
   }
 
+#define GRUB_EFI_IMAGE_SECURITY_DATABASE_GUID \
+  { 0xd719b2cb, 0x3d3a, 0x4596, \
+    { 0xa3, 0xbc, 0xda, 0xd0, 0x0e, 0x67, 0x65, 0x6f } \
+  }
+
 #define GRUB_EFI_MEMORY_TYPE_INFORMATION_GUID \
   { 0x4c19049f, 0x4137, 0x4dd3, \
     { 0x9c, 0x10, 0x8b, 0x97, 0xa8, 0x3f, 0xfd, 0xfa } \
-- 
2.36.1

